File: 2018/david_pivazyan/Singleton/Singleton.cpp
#include "Singleton.h"

Singleton* Singleton::instance = 0;

Singleton* Singleton::getInstance()
{
	if (instance == 0)                  //Meyer's Singleton!
	{									//
		instance = new Singleton();		//
	}									//
	return instance;}					//

Singleton::Singleton()
{}

File: 2019/alex_gasparyan/Singleton.java
class Test {
	public static void main(String[] args) {
		MySingleton singleton = MySingleton.getInstance();
	}
}

class MySingleton {

    private MySingleton() {
        
    }

    private static class Holder {
        private static MySingleton instance = new MySingleton();
    }

    public static MySingleton getInstance() {
        return Holder.instance;
    }
}